### PR TITLE
bugfix: read/write fd dropped unexpectedly.

### PR DESCRIPTION
**Pull request by WeiZhang555**
read/write function invokes

```
self.check_fd_flags(data, f.as_raw_fd(), flags)?;
```

which takes ownership of `data` and it's inner `File`, after the function call returns, File will be dropped and cannot be read/write later anymore, causing a failed read/write.

This bug is triggered when `no_open` config option is set, in this case, `self.get_data` opens a new file and it's dropped immediately after `check_fd_flags`.

The bug can be solved by create a new clone of data by `data.clone()`, the original data(and inner File) can be kept in whole lifetime of read/write function.

### CODE REVIEW

**Review by eryugey — APPROVED** [review]

File: src/passthrough/mod.rs [hunk 1]
```diff
@@ -1479,4 +1483,144 @@ mod tests {
         assert!(!fs.cfg.no_open);
         assert!(!fs.cfg.writeback);
     }
+
+    impl ZeroCopyReader for File {
```
Comment (bergwolf) [inline src/passthrough/mod.rs:1487]: Why are these needed? The commit message does not seem to explain it.

Comment (WeiZhang555) [reply src/passthrough/mod.rs:1487]: This resides in `mod test`, at https://github.com/cloud-hypervisor/fuse-backend-rs/pull/184/files/9ca710da29796b759d6d782e8dd2d9209d5bf838#diff-7f83e7da9ae6f4daf23fadb16185e191349a8f96620bd58d29483e3b28ec1a1cR1593

I need a `&mut dyn ZeroCopyReader`, in fuse-backend-rs, I didn't find a convenient implementation of `ZeroCopyReader` and `ZeroCopyWriter`, so I have to implement one in `mod test` for test usage.

**Review by bergwolf — APPROVED** [review]
